### PR TITLE
fix(docs-infra): fix redirect in angular version selector

### PR DESCRIPTION
**Pull request by sonukapoor**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [X] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [X] Tests for the changes have been added (for bug fixes / features)
- [X] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [X] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: #35630

## Does this PR introduce a breaking change?

- [ ] Yes
- [X] No

### CODE REVIEW

File: aio/src/app/app.component.ts [hunk 1]
```diff
@@ -149,7 +149,7 @@ export class AppComponent implements OnInit {
     ]).subscribe(([versionInfo, versions]) => {
       // TODO(pbd): consider whether we can lookup the stable and next versions from the internet
       const computedVersions: NavigationNode[] = [
-        { title: 'next', url: 'https://next.angular.io' },
+        { title: 'next', url: 'https://next.angular.io/' },
         { title: 'stable', url: 'https://angular.io' },
```
Comment (ericmartinezr) [inline aio/src/app/app.component.ts]: You should change both, as both have the same problem.

**Review by gkalpak — CHANGES_REQUESTED** [review]
From https://github.com/angular/angular/pull/35318#discussion_r379393795:

> Currently, all `docVersions` URLs contains a trailing `/`, but if we wanted to be cautious, we should check (and if necessary add a trailing `/`).

And that's why one should always err on the side of caution :grin:

##
So, in addition to the current changes, I still believe it is better to ensure there is always exactly one `/` between `version.url` and `currentUrl`:

https://github.com/angular/angular/blob/e084835fb111147ac7497036d38a0b5c012d5cd6/aio/src/app/app.component.ts#L235

**Comment by gkalpak** [comment]
Also, please make sure the commit message follows our [guidelines][1] (e.g. in this case include a `Fixes #35630` footer).

[1]: https://github.com/angular/angular/blob/e084835fb/CONTRIBUTING.md#commit

File: aio/src/app/app.component.ts [hunk 2]
```diff
@@ -232,7 +232,8 @@ export class AppComponent implements OnInit {
   onDocVersionChange(versionIndex: number) {
     const version = this.docVersions[versionIndex];
     if (version.url) {
-      this.locationService.go(`${version.url}${this.currentUrl}`);
+      const versionUrl = version.url  + (!version.url.endsWith('/') ? '/' : '');
```
Comment (gkalpak) [inline aio/src/app/app.component.ts]: :+1:
Can you also add a test for this?

File: aio/src/app/app.component.spec.ts [hunk 3]
```diff
@@ -417,6 +417,13 @@ describe('AppComponent', () => {
         selectElement.triggerEventHandler('change', { option: versionWithoutUrl, index: versionWithoutUrlIndex });
         expect(locationService.go).not.toHaveBeenCalled();
       });
+
+      it('should navigate when change to a version with a url that does not end with `/`', async () => {
+        const versionWithoutSlashIndex = component.docVersions.length;
+        const versionWithoutSlashUrl = component.docVersions[versionWithoutSlashIndex] = { url: 'https://next.angular.io', title: 'foo' };
+        selectElement.triggerEventHandler('change', { option: versionWithoutSlashUrl, index: versionWithoutSlashIndex });
+        expect(locationService.go).not.toHaveBeenCalledWith('https://next.angular.io/');
```
Comment (gkalpak) [inline aio/src/app/app.component.spec.ts]: Can you check with a path (to better highlight what the problem would be if no `/` were added)?

Comment (sonukapoor) [reply aio/src/app/app.component.spec.ts]: Sure. Will take a look.

Comment (sonukapoor) [reply aio/src/app/app.component.spec.ts]: @gkalpak All done. There are some strange CI errors, that I am not too sure about.

File: aio/src/app/app.component.spec.ts [hunk 4]
```diff
@@ -417,6 +417,14 @@ describe('AppComponent', () => {
         selectElement.triggerEventHandler('change', { option: versionWithoutUrl, index: versionWithoutUrlIndex });
         expect(locationService.go).not.toHaveBeenCalled();
       });
+
+      it('should navigate when change to a version with a url that does not end with `/`', async () => {
+        locationService.urlSubject.next('https://angular.io/docs#section-1');
```
Comment (gkalpak) [inline aio/src/app/app.component.spec.ts]: `locationService.urlSubject` is supposed to include the path part only (see other occurrences of `urlSubject.next()` in this file).

**Review by gkalpak — CHANGES_REQUESTED** [review]

File: aio/src/app/app.component.spec.ts (hunk 4 above)
Comment (sonukapoor) [reply aio/src/app/app.component.spec.ts]: My bad. I fixed it. The test was actually not accurate.

**Review by gkalpak — APPROVED** [review]
Thank you, sonukapoor ✨

**Pull request closed without merging**